crates/katana/rpc/rpc/tests/torii.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+
+use dojo_test_utils::sequencer::{get_default_test_starknet_config, TestSequencer};
+use jsonrpsee::http_client::HttpClientBuilder;
+use katana_core::sequencer::SequencerConfig;
+use katana_rpc_api::katana::KatanaApiClient;
+use katana_rpc_api::torii::ToriiApiClient;
+use katana_rpc_types::transaction::{TransactionsPage, TransactionsPageCursor};
+use starknet::accounts::{Account, Call};
+use starknet::core::types::FieldElement;
+use starknet::core::utils::get_selector_from_name;
+use tokio::time::sleep;
+
+use crate::common::prepare_contract_declaration_params;
+
+mod common;
+
+pub const ENOUGH_GAS: &str = "0x100000000000000000";
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_get_transactions() {
+    let sequencer = TestSequencer::start(
+        SequencerConfig { block_time: None, no_mining: true, ..Default::default() },
+        get_default_test_starknet_config(),
+    )
+    .await;
+
+    let client = HttpClientBuilder::default().build(sequencer.url()).unwrap();
+
+    let account = sequencer.account();
+
+    let path: PathBuf = PathBuf::from("tests/test_data/cairo1_contract.json");
+    let (contract, compiled_class_hash) = prepare_contract_declaration_params(&path).unwrap();
+    let contract = Arc::new(contract);
+
+    // Should return successfully when no transactions have been mined.
+    let cursor = TransactionsPageCursor { block_number: 0, transaction_index: 0 };
+
+    let response: TransactionsPage = client.get_transactions(cursor).await.unwrap();
+
+    assert!(response.transactions.is_empty());
+    assert!(response.cursor.block_number == 1);
+    assert!(response.cursor.transaction_index == 0);
+
+    let declare_res = account.declare(contract.clone(), compiled_class_hash).send().await.unwrap();
+
+    // Should return successfully with single pending txn.
+    let response: TransactionsPage = client.get_transactions(response.cursor).await.unwrap();
+
+    assert!(response.transactions.len() == 1);
+    assert!(response.cursor.block_number == 1);
+    assert!(response.cursor.transaction_index == 1);
+
+    // Create block 1.
+    let _: () = client.generate_block().await.unwrap();
+
+    // Should properly increment to new empty pending block
+    let response: TransactionsPage = client.get_transactions(response.cursor).await.unwrap();
+
+    assert!(response.transactions.is_empty());
+    assert!(response.cursor.block_number == 2);
+    assert!(response.cursor.transaction_index == 0);
+
+    // Should block on cursor at end of page and return on new txn
+    let long_poll_future = client.get_transactions(response.cursor);
+
+    // Yield the current task, allowing the long poll to be established.
+    tokio::task::yield_now().await;
```

Comment:
afaik this doesn't do anything

Reply:
i think calling `tokio::task::spawn` on `long_poll_future` and awaiting on the join handle after the `deploy_txn_future` future finish, would perhaps be a better way to showcase your intention